lib/Units.pm

```diff
@@ -1,998 +1,789 @@
 	# ENGLISH LENGTHS
-	# in    -- inch
-	# ft    -- feet
-	# mi    -- mile
-	# furlong -- (FFF system) 0.125 mile
-	# light-year
-	# AU	-- Astronomical Unit
-	# parsec
-	#
-	'in' => {
-		'factor' => 0.0254,
-		'm'      => 1
-	},
-	'inch' => {
-		'factor' => 0.0254,
-		'm'      => 1
-	},
-	'inches' => {
-		'factor' => 0.0254,
-		'm'      => 1
-	},
-	'ft' => {
-		'factor' => 0.3048,
-		'm'      => 1
-	},
-	'feet' => {
-		'factor' => 0.3048,
-		'm'      => 1
-	},
-	'foot' => {
-		'factor' => 0.3048,
-		'm'      => 1
-	},
-	'mi' => {
-		'factor' => 1609.344,
-		'm'      => 1
-	},
-	'furlong' => {
-		'factor' => 201.168,
-		'm'      => 1
-	},
-	'light-year' => {
-		#'factor'    => 9.46E15,
-		'factor' => 9460730472580800,
-		'm'      => 1
-	},
-	'AU' => {
-		'factor' => 149597870700,
-		'm'      => 1
-	},
-	'parsec' => {
-		'factor' => 3.08567758149137E16,    #30.857E15,
-		'm'      => 1
-	},
-	# VOLUME
-	# L   -- liter
-	# ml -- milliliters
-	# cc -- cubic centermeters
-	# dL  -- deci-liter
-	#
-	'L' => {
-		'factor' => 0.001,
-		'm'      => 3
-	},
-	'cc' => {
-		'factor' => 1E-6,
-		'm'      => 3,
-	},
-	'ml' => {
-		'factor' => 1E-6,
-		'm'      => 3,
-	},
-	'dL' => {
-		'factor' => 0.0001,
-		'm'      => 3
-	},
-	'cup' => {
-		'factor' => 0.000236588,
-		'm'      => 3
-	},
-	'cups' => {
-		'factor' => 0.000236588,
-		'm'      => 3
-	},
-	# VELOCITY
-	# knots -- nautical miles per hour
-	# c		-- speed of light
-	#
-	'knots' => {
-		'factor' => 0.5144444444,
-		'm'      => 1,
-		's'      => -1
-	},
-	'c' => {
-		'factor' => 299792458,    # exact
-		'm'      => 1,
-		's'      => -1
-	},
-	'mph' => {
-		'factor' => 0.44704,
-		'm'      => 1,
-		's'      => -1
-	},
-	# MASS
-	# mg   -- milligrams
-	# g    -- grams
-	# kg   -- kilograms
-	# tonne -- metric ton
-	#
-	'mg' => {
-		'factor' => 0.000001,
-		'kg'     => 1
-	},
-	'g' => {
-		'factor' => 0.001,
-		'kg'     => 1
-	},
-	'tonne' => {
-		'factor' => 1000,
-		'kg'     => 1
+	inch => {
+		factor  => 0.0254,
+		m       => 1,
+		aliases => [ 'inches', 'in' ]
+	},
+	foot => {
+		factor  => 0.3048,
+		m       => 1,
+		aliases => [ 'feet', 'ft' ]
+	},
+	mile => {
+		factor  => 1609.344,
+		m       => 1,
+		aliases => [ 'miles', 'mi' ]
 	},
+	furlong => {    # (FFF system) 0.125 mile
+		factor => 201.168,
+		m      => 1
+	},
+	'light-year' => {    # 9.46E15,
+		factor => 9460730472580800,
+		m      => 1
+	},
+	AU => {              # Astronomical Unit
+		factor => 149597870700,
+		m      => 1
+	},
+	parsec => {          # 30.857E15,
+		factor => 3.08567758149137E16,
+		m      => 1
+	},
+
+	# VOLUME: fundamental unit m^3 (cubic meter)
+	L => {               # liter
+		factor => 0.001,
+		m      => 3
+	},
+	ml => {              # milliliter (cubic centimeter)
+		factor  => 1E-6,
+		m       => 3,
+		aliases => ['cc']
+	},
+	dL => {              # deciliter
+		factor => 0.0001,
+		m      => 3
+	},
+	cup => {
+		factor => 0.000236588,
+		m      => 3
+	},
+	cups => {
+		factor => 0.000236588,
+		m      => 3
+	},
```

Comment:
You probably want to make `cups` an alias of `cup`

Reply:
I missed `cups` and `dalton`.  Thanks for pointing those out.

Yeah, unnecessary quoting with fat commas bugs me.  I couldn't leave it.